### PR TITLE
Fix Glide dependencies and pickup golang 1.9.2

### PR DESCRIPTION
**Pull request by bassam**

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -20,88 +20,33 @@ imports:
...
-  - pkg/transport
-  - pkg/types
-  - pkg/wait
-  - raft
-  - raft/raftpb
-  - rafthttp
-  - snap
-  - snap/snappb
-  - store
   - version
```
Comment (travisn) [inline glide.lock]: we can get rid of this etcd version package, just need to delete an obsolete line where it's called in `cmd/rook/version.go`

File: glide.yaml [hunk 2]
```diff
@@ -10,13 +10,6 @@ excludeDirs:
 - Documentation
 - images
 import:
```
Comment (travisn) [inline glide.yaml]: not adding the operator-kit dependency now? i'll add it with my change

File: glide.lock (hunk 1 above)
Comment (kokhang) [reply glide.lock]: +1

File: glide.yaml [hunk 3]
```diff
@@ -25,38 +18,44 @@ import:
...
-- package: k8s.io/apiextensions-apiserver
-  version: 3eb4c5fefbe7ad866747f78c7e4994cb555170c6
 - package: github.com/prometheus/client_golang
   version: e7e903064f5e9eb5da98208bae10b475d4db0f8c
   subpackages:
   - prometheus
   - prometheus/promhttp
 - package: github.com/jbw976/go-ps
   version: 82859aed1b5dafbbe6ad4f3dac5404d2a742cf18
+# Kuberenetes dependencies. These must all be updated together
```
Comment (kokhang) [inline glide.yaml]: I like how you group these together. Awesome

File: glide.yaml [hunk 4]
```diff
@@ -25,38 +18,44 @@ import:
...
+  version: be41f5093e2b05c7a0befe35b04b715eb325ab43
+- package: k8s.io/apimachinery
+  version: 728b986edb9bd4bb8a022a8cc13573a97ecc2239
+- package: k8s.io/client-go
+  version: 42a124578af9e61f5c6902fa7b6b2cb6538f17d2
+- package: k8s.io/api
+  version: e24ed681f56165befbea50a24f7e9f1c079ac366
+- package: k8s.io/kubernetes
+  version: df7f4b3526a580ef122aaeef61ab52842a60a88b
+#testImport: imports only used during testing go below
```
Comment (kokhang) [inline glide.yaml]: you can still use testImport for test packages that dont conflict with `import`

Comment (bassam) [reply glide.yaml]: its best we avoid it all together. `dep` doesn't have it, and the glide folks don't seem to have a solution for it. see referenced issue.

File: glide.lock (hunk 1 above)
Comment (bassam) [reply glide.lock]: ok

**Comment by bassam** [comment]
@jbw976 I doubt it. Go includes files used at the package level. we were only using etcd.version.

**Pull request merged**